File: src/main/java/com/example/indie91/Models/Content.java
package com.example.indie91.Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "content")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Content {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(name = "content_type")
    private String contentType;

    private String url;
    private String description;

    @Column(name = "is_active")
    private boolean isActive;

    @Column(name = "views_count")
    private Integer viewsCount;

    @Column(name = "likes_count")
    private Integer likesCount;

    @Column(name = "share_count")
    private Integer shareCount;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "influencer_id", nullable = false)
    private InfluencerProfile influencer;

    @Column(name = "tag_id")
    private UUID tagId;

}

File: models/content.go
package models

import "time"

type Content struct {
	ID                string    `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	InfluencerID      string    `gorm:"type:uuid" json:"influencer_id"`
	ContentType       string    `gorm:"type:varchar(5);default:IMAGE" json:"content_type"`
	URL               string    `json:"url"`
	CreatedAt         time.Time `gorm:"default:now()" json:"created_at"`
	UpdatedAt         time.Time `gorm:"default:now()" json:"updated_at"`
	IsActive          bool      `gorm:"default:true" json:"is_active"`
	ViewsCount        int       `gorm:"default:0" json:"views_count"`
	LikesCount        int       `gorm:"default:0" json:"likes_count"`
	TagID             *string   `gorm:"type:uuid" json:"tag_id"`
	Description       string    `json:"description"`
	ShareCount        int       `json:"share_count"`
}
